samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -26,7 +26,7 @@
 
 public abstract class AbstractApplicationRunner implements ApplicationRunner {
```

Comment:
> Yes you did: #74

Umm..My bad.. I didn't realize ExecutionEnvironment PR is same as the one ApplicationRunner introduced. Anyway, I have some questions now.

> since all the Runners take config as a constructor parameter, 

Is that assumption right? Because if we expect all Runner implementations to contain state (as in Config), then why not just have ApplicationRunner as an abstract class? We can fully get rid of the interface and use a final method for streamFromConfig. 

If we do decide to go with an interface for ApplicationRunner, it is pretty misleading for the user to call streamFromConfig when the expected parameter is streamId.